### PR TITLE
feat: 'depends' performance

### PR DESCRIPTION
**Pull request by kgilpin**
* Optimize target file enumeration
* Cache file modified times
* Remove the most verbose logging
* Remove source locations that are blank

Before optimization:
```
node ./built/cli.js depends -d ~/source/land-of-apps/forem > forem.log  11.41s user 6.40s system 169% cpu 10.537 total
```

After optimization:
```
node ./built/cli.js depends -d ~/source/land-of-apps/forem > forem.log  3.86s user 1.59s system 127% cpu 4.296 total
```

### CODE REVIEW

File: packages/cli/src/utils.ts [hunk 1]
```diff
@@ -91,18 +92,61 @@ export async function processFiles(
...
+    for (const childName of await readdir(dir)) {
+      const child = await stats(join(dir, childName));
+      if (child && child.isDirectory()) {
+        await processDir(join(dir, childName));
+      }
+    }
+  };
+  await processDir(baseDir);
+
+  if (matchCount) await q.drain();
```
Comment (dividedmind) [inline packages/cli/src/utils.ts:147]: Perhaps it'd be better to check `q.idle()` instead — I think `q.drain()` might hang if the queue has emptied in the meantime.

File: packages/cli/tests/unit/depends.spec.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ describe('Depends', () => {
   beforeAll(async () => verbose(process.env.DEBUG === 'true'));
   beforeEach(async () => {
     appMapDir = tmp.dirSync().name;
-    expected = join(appMapDir, 'user_page_scenario').replace(/\\/g, '/');
+    expected = join(appMapDir, 'user_page_scenario');
```
Comment (dividedmind) [inline packages/cli/tests/unit/depends.spec.ts:23]: I assume this was so the tests pass on Windows correctly. I wonder what changed that it's not needed anymore?

**Review by dividedmind — APPROVED** [review]
Looks good, but I'd pay attention to that queue so it doesn't end up hanging.

File: packages/cli/tests/unit/depends.spec.ts (hunk 2 above)
Comment (kgilpin) [reply packages/cli/tests/unit/depends.spec.ts:23]: Before, we were using `glob` to enumerate the files. Now we are walking the directory tree ourselves, so the file paths don't require this gsub.

File: packages/cli/src/utils.ts (hunk 1 above)
Comment (kgilpin) [reply packages/cli/src/utils.ts:147]: Yeah this has always been an issue with these queues. I'll change it `drain unless q.idle()`

Comment (kgilpin) [reply packages/cli/src/utils.ts]: I've done this project-wide, let me know what you think.

File: packages/cli/tests/unit/queue.spec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+describe('async', () => {
+  it('hangs if you drain an idle queue', async () => {
+    const q = queue(async () => {}, 1);
+    expect(q.idle()).toBe(true);
+    let drained = false;
+    q.drain().then(() => (drained = true));
+    setTimeout(() => {
+      assert(!drained, 'drain should not have resolved yet');
+    }, 50);
```
Comment (dividedmind) [inline packages/cli/tests/unit/queue.spec.ts:12]: This timeout will fire after the test case returns. I suggest instead `await`ing a `setTimeout` from `timers/promises` and then asserting after that.

**Pull request merged**